src/modules/75-cf/components/GitSyncModal/GitSyncModal.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useCallback, useRef, useState } from 'react'
+import { FontVariation, Color } from '@harness/design-system'
+import type { FormikProps } from 'formik'
+import * as Yup from 'yup'
+import { useParams } from 'react-router-dom'
+import { Button, ButtonVariation, Formik, FormikForm, Heading, Layout, ModalDialog } from '@harness/uicore'
+import { useToaster } from '@common/exports'
+import { useCreateGitRepo } from 'services/cf'
+import { fullYamlPathRegex } from '@common/utils/StringUtils'
+import { getErrorMessage } from '@cf/utils/CFUtils'
+import GitSyncSetupButton from '@cf/components/GitSyncSetupButton/GitSyncSetupButton'
+import { useStrings } from 'framework/strings'
+import { GitSyncForm, GitSyncFormFields } from '@gitsync/components/GitSyncForm/GitSyncForm'
+
+export const GitSyncModal: React.FC = () => {
+  const { orgIdentifier, projectIdentifier, accountId: accountIdentifier } = useParams<Record<string, string>>()
+  const { showError } = useToaster()
+  const { getString } = useStrings()
+  const formikRef = useRef<FormikProps<GitSyncFormFields>>()
+
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const hideModal = useCallback(() => setIsOpen(false), [])
+  const openModal = useCallback(() => setIsOpen(true), [])
+
+  const { mutate: createGitRepo } = useCreateGitRepo({
+    identifier: projectIdentifier,
+    queryParams: {
+      accountIdentifier,
+      orgIdentifier
+    }
+  })
+
+  const handleSubmit = useCallback(
+    async (formValues: GitSyncFormFields) => {
+      // get file name and root folder path from filePath field
+      const fullPath = formValues?.filePath
+      const pathNodes = fullPath?.split('/').filter((node: string) => !!node) // remove empty
+      const yamlFileName = '/' + pathNodes?.pop()
+      const yamlFilePath = pathNodes?.length ? '/' + pathNodes?.join('/') + '/' : ''
+
+      const requestData = {
+        repoIdentifier: formValues.repo,
+        repoReference: formValues.repo,
+        branch: formValues.branch,
+        filePath: yamlFileName,
+        rootFolder: yamlFilePath,
+        connectorReference: formValues.connectorRef?.value
```

Comment:
Should this be
```suggestion
        connectorReference: formValues.connectorRef?.valueOf
```
'value' does not seem to exist on type 'string | ConnectorSelectedValue'

Reply:
Fixed on latest commit